docs/components/combobox/index.md

```diff
@@ -0,0 +1,254 @@
+  as |combobox|
+>
+  <combobox.Option>
+    <!-- The content of each popover list item will be rendered here -->
+    {{combobox.option}}
+  </combobox.Option>
+</Form::Controls::Combobox>
+```
+
+```js
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+
+export default class extends Component {
+  @tracked selected;
+}
+```
+
+## onChange
+
+Called when the user makes a selection. It is called with the selected option (derived from `@options`) as its only argument. You'll want to update `@selected` with the new value in your on change handler.
+
+```hbs
+<Form::Controls::Combobox
+  @onChange={{this.handleChange}}
+  @options={{this.options}}
+  @selected={{this.selected}}
+  as |combobox|
+>
+  <combobox.Option>
+    {{combobox.option}}
+  </combobox.Option>
+</Form::Controls::Combobox>
+```
+
+```js
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+export default class extends Component {
+  @tracked selected;
+
+  options = ['Blue', 'Red', 'Yellow'];
+
+  @action
+  handleChange(option) {
+    this.selected = option;
+  }
+}
+```
+
+## Option Key
+
+Optional.
+
+The `@optionKey` argument is used when your `@options` take the shape of an array of objects. The `@optionKey` is used to determine two things internally:
+
+1. The displayed value inside of the input of the combobox
+2. Used as the key in the default filtering scenario where we filter `@options`. To properly filter the `@options` based on the user input from the textbox, we need to know how to compare the entered value to each object. The `@optionKey` tells us which key of the object to use for this filtering.
+
+In the example below, we set `@optionKey='label'`. Our `@options` objects have a `label` key and we want the label of the selected option to be used for the selected value, as well as for filtering as the user types.
+
+```hbs
+<Form::Controls::Combobox
+  @onChange={{this.handleChange}}
+  @options={{this.options}}
+  @optionKey='label'
+  @selected={{this.selected}}
+  as |combobox|
+>
+  <combobox.Option>
+    {{combobox.option}}
+  </combobox.Option>
+</Form::Controls::Combobox>
+```
+
+```js
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+export default class extends Component {
+  @tracked selected;
+
+  options = [
+    {
+      label: 'Blue',
+      value: 'blue',
+    },
+    {
+      label: 'Green',
+      value: 'green',
+    },
+    {
+      label: 'Yellow',
+      value: 'yellow',
+    },
+    {
+      label: 'Orange',
+      value: 'orange',
+    },
+    {
+      label: 'Red',
+      value: 'red',
+    },
+    {
+      label: 'Purple',
+      value: 'purple',
+    },
+    {
+      label: 'Teal',
+      value: 'teal',
+    },
+  ];
+
+  @action
+  handleChange(option) {
+    this.selected = option;
+  }
+}
+```
+
+## onFilter
```

Comment:
This can be marked as optional, right? Maybe also point out what the default filtering logic does, so it's clear you don't need this if the defaults work as well.

Reply:
Just wrote some more as part of https://github.com/CrowdStrike/ember-toucan-core/pull/200/commits/2b74de095b48c18335903e0aabaa25d5717cc734, thank you!